neo4j/templates/_helpers.tpl

```diff
@@ -323,3 +328,28 @@ affinity:
     {{- end }}
     {{- end }}
 {{- end -}}
+
+{{- define "neo4j.secretName" -}}
+    {{- if .Values.neo4j.passwordFromSecret -}}
+        {{- $secret := (lookup "v1" "Secret" .Release.Namespace .Values.neo4j.passwordFromSecret) }}
+        {{- $secretExists := $secret | all }}
+        {{- if not ( $secretExists ) -}}
+            {{ fail (printf "Secret %s configured in 'neo4j.passwordFromSecret' not found" .Values.neo4j.passwordFromSecret) }}
+        {{- else if not (hasKey $secret.data "NEO4J_AUTH") -}}
+            {{ fail (printf "Secret %s must contain key NEO4J_DATA" .Values.neo4j.passwordFromSecret) }}
+        {{- else if not (index $secret.data "NEO4J_AUTH" | b64dec | regexFind "^neo4j\\/\\w*") -}}
+            {{ fail (printf "Password in secret %s must start with the characters 'neo4j/'" .Values.neo4j.passwordFromSecret) }}
+        {{- else -}}
+            {{- printf "%s" (tpl .Values.neo4j.passwordFromSecret $) -}}
+         {{- end -}}
+    {{- else -}}
+        {{- printf "%s-auth" (include "neo4j.fullname" .) -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "neo4j.passwordWarning" -}}
```

Comment:
i think it would be good to add a unit test case to check the password warning

Reply:
We can't unit test that because template doesn't output the notes. We could add an integration test

Reply:
okay...i believe the current integration test uses the passwordFromSecret and not password anymore..so adding an extra integration test just to check the warning sounds overkill too me...if we can check it as part of existing integration test its okay otherwise its fine

Reply:
Yes I checked the output in one of the other auth tests